managed/services/management/node.go

```diff
@@ -121,9 +131,16 @@ func (s *NodeService) Register(_ context.Context, req *managementpb.RegisterNode
 			CreateNodeExporter(tx.Querier, pmmAgent.AgentID, nil, isPushMode(req.MetricsMode), req.DisableCollectors,
 				pointer.ToStringOrNil(req.AgentPassword), "")
 		return err
-	}); e != nil {
+	})
+	if e != nil {
 		return nil, e
 	}
 
+	apiKeyName := fmt.Sprintf("pmm-agent-%s-%d", req.NodeName, rand.Int63())
+	_, res.Token, e = s.akp.CreateAdminAPIKey(ctx, apiKeyName)
+	if e != nil {
+		return res, e
```

Comment:
Is it intended we return `res` as not `nil`?

Reply:
I think it's better to return nil